azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/PromptWrapper.java

```diff
@@ -0,0 +1,276 @@
+                options, defaultEntity, getNameFunc, isRequired);
+    }
+
+    public <T> List<T> handleMultipleCase(String templateId, List<T> options, Function<T, String> getNameFunc)
+            throws IOException, NoResourcesAvailableException {
+        final Map<String, Object> variables = createVariableTables(templateId);
+        final boolean allowEmpty = TemplateUtils.evalBoolean("allow_empty", variables);
+        if (options.size() == 0) {
+            if (!allowEmpty) {
+                throw new NoResourcesAvailableException(TemplateUtils.evalText("message.empty_options", variables));
+            } else {
+                final String warningMessage = TemplateUtils.evalText("message.empty_options", variables);
+                if (StringUtils.isNotBlank(warningMessage)) {
+                    log.warn(warningMessage);
+                }
+            }
+            return options;
+        }
+        final boolean autoSelect = TemplateUtils.evalBoolean("auto_select", variables);
+        final boolean defaultSelected = TemplateUtils.evalBoolean("default_selected", variables);
+        if (options.size() == 1) {
+            if (autoSelect) {
+                log.info(TemplateUtils.evalText("message.auto_select", variables));
+                return options;
+            } else {
+                if (!this.prompt.promoteYesNo(TemplateUtils.evalText("promote.one", variables), defaultSelected, false)) {
+                    // user cancels
+                    final String warningMessage = TemplateUtils.evalText("message.select_none", variables);
+                    if (StringUtils.isNotBlank(warningMessage)) {
+                        log.warn(warningMessage);
+                    }
+                    return Collections.emptyList();
+                }
+                return options;
+            }
+        }
+        final List<T> selectedEntities = prompt.promoteMultipleEntities(TemplateUtils.evalText("promote.header", variables),
+                TemplateUtils.evalText("promote.many", variables),
+                TemplateUtils.evalText("promote.header", variables), options, getNameFunc, allowEmpty,
+                defaultSelected ? "to select ALL" : "to select NONE", defaultSelected ? options : Collections.emptyList());
+        if (selectedEntities.isEmpty()) {
+            final String warningMessage = TemplateUtils.evalText("message.select_none", variables);
+            if (StringUtils.isNotBlank(warningMessage)) {
+                log.warn(warningMessage);
+            }
+        }
+        return selectedEntities;
+    }
+
+    public String handle(String templateId, boolean autoApplyDefault) throws InvalidConfigurationException, IOException, ExpressionEvaluationException {
+        return handle(templateId, autoApplyDefault, null);
+    }
+
+    public String handle(String templateId, boolean autoApplyDefault, Object defaultValueCli)
```

Comment:
defaultValueCli is a bit confused, shall we change it to `cliParameter` or some word more clearly?

Reply:
fixed